### DISCUSSION TITLE
how to make dependencies tracked in the dvc.yaml

### DISCUSSION THREAD

**Discussion by Ichiruchan** [treeverse/dvc#8085] [Help]
Hello, I'am new at dvc. I am using spacy which is a nlp framework and has script to generate dvc.yaml automatically. After generating yaml file, I executed `dvc repro` to re-product my pipeline and then executed `dvc push` to push data to my remote storage. And I find that only files in outs part will be pushed. But I also want to push the first dependency(the original data) to my remote storage. Right now, my solution is `dvc add first_dependency` and `dvc push`. Is there any way more advanced? Thank you in advance for your reply.

**Comment by pmrowla** [comment]
Pipeline stage dependencies are not tracked, only outputs. What you are doing already is correct, data dependencies  (that are not outputs of any other pipeline stage) should be tracked with `dvc add`
